alerts/session_opened_sensitive_user.py

```diff
@@ -45,25 +43,29 @@ def main(self):
             QueryStringMatch('summary:"session opened"'),
         ])
 
-        search_query.add_must_not([
-            RangeMatch('utctimestamp', r1, r2)
-        ])
+        for expectedtime in self._config['scan_expected'].values():
+            r1 = datetime.datetime.now().replace(hour=int(expectedtime['start_hour']), minute=int(expectedtime['start_minute']), second=int(expectedtime['start_second'])).isoformat()
+            r2 = datetime.datetime.now().replace(hour=int(expectedtime['end_hour']), minute=int(expectedtime['end_minute']), second=int(expectedtime['end_second'])).isoformat()
+            search_query.add_must_not([
+                RangeMatch('utctimestamp', r1, r2)
+            ])
+
         search_query.add_must(superquery)
 
         self.filtersManual(search_query)
-        self.searchEventsAggregated('details.hostname', samplesLimit=10)
+        self.searchEventsAggregated('details.program', samplesLimit=10)
         self.walkAggregations(threshold=1)
 
     def onAggregation(self, aggreg):
         category = 'session'
         severity = 'WARNING'
         tags = ['pam', 'syslog']
 
-        prog = ''
-        for p in aggreg['events']:
-            if p['_source']['details']['program'] != prog:
-                prog += p['_source']['details']['program']
+        uniquehosts = []
+        for e in aggreg['events']:
+            if e['_source']['details']['hostname'] not in uniquehosts:
+                uniquehosts.append(e['_source']['details']['hostname'])
 
-        summary = '{0} session opened for scanning user outside of the expected window on {1} [{2}]'.format(prog, aggreg['value'], aggreg['count'])
+        summary = 'Session opened by the scan user outside of the expected window - sample hosts: {0} [total {1} hosts]'.format(' '.join(uniquehosts), aggreg['count'])
```

Comment:
@mpurzynski one comment here actually, we should probably change this from "scan" to sensitive user